libbeat/common/field.go

```diff
@@ -94,6 +105,23 @@ func (d *DynamicType) Unpack(s string) error {
 	return nil
 }
 
+// Unpack checks if objectTypeParams are mixed with top level objectType configuration
+// and creates Field out of config
+func (f *Field) Unpack(c *Config) error {
+	type tmpField Field
+	var tf tmpField
+	if err := c.Unpack(&tf); err != nil {
+		return err
+	}
+
+	if len(tf.ObjectTypeParams) != 0 && (tf.ScalingFactor != 0 || tf.ObjectTypeMappingType != "" || tf.ObjectType != "") {
+		return errors.New("mixing top level objectType configuration with array of object type configurations is forbidden")
+	}
```

Comment:
Should this check be done in a `Validate()` method?

Reply:
I wasn't aware of the `Validate` method - agree that this fits better, will change.